File: app/src/main/java/com/hjc/base/ui/senior/activity/MotionActivity.java
package com.hjc.base.ui.senior.activity;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.view.View;

import androidx.annotation.Nullable;

import com.alibaba.android.arouter.facade.annotation.Route;
import com.hjc.base.R;
import com.hjc.base.constant.RoutePath;
import com.hjc.base.databinding.ActivityMotionBinding;
import com.hjc.base.utils.helper.RouteManager;
import com.hjc.baselib.activity.BaseActivity;
import com.hjc.baselib.viewmodel.CommonViewModel;

@Route(path = RoutePath.URL_MOTION)
public class MotionActivity extends BaseActivity<ActivityMotionBinding, CommonViewModel> {

    @Override
    protected int getLayoutId() {
        return R.layout.activity_motion;
    }

    @Override
    protected CommonViewModel getViewModel() {
        return null;
    }

    @Override
    protected void initData(@Nullable Bundle savedInstanceState) {
    }

    @Override
    protected void addListeners() {
        mBindingView.setOnClickListener(this);

        mBindingView.titleBar.setOnViewLeftClickListener(view -> finish());
    }

    @SuppressLint("NonConstantResourceId")
    @Override
    protected void onSingleClick(View v) {
        switch (v.getId()) {
            case R.id.btn1:
                RouteManager.jump(RoutePath.URL_TOUCH);
                break;

            case R.id.btn2:
                RouteManager.jump(RoutePath.URL_VIEW);
                break;

            default:
                break;
        }
    }
}

File: module-senior/src/main/java/com/hjc/module_senior/ui/motion/MotionActivity.kt
package com.hjc.module_senior.ui.motion

import android.os.Bundle
import android.view.View
import com.alibaba.android.arouter.facade.annotation.Route
import com.gyf.immersionbar.ImmersionBar
import com.hjc.library_base.activity.BaseActivity
import com.hjc.library_base.viewmodel.CommonViewModel
import com.hjc.library_common.router.RouteManager
import com.hjc.library_common.router.RoutePath
import com.hjc.library_widget.bar.OnViewLeftClickListener
import com.hjc.module_senior.R
import com.hjc.module_senior.databinding.SeniorActivityMotionBinding

@Route(path = RoutePath.Senior.URL_MOTION)
class MotionActivity : BaseActivity<SeniorActivityMotionBinding, CommonViewModel>() {

    override fun getLayoutId(): Int {
        return R.layout.senior_activity_motion
    }

    override fun createViewModel(): CommonViewModel? {
        return null
    }

    override fun getImmersionBar(): ImmersionBar? {
        return ImmersionBar.with(this)
            .fitsSystemWindows(true)
            .statusBarColor(R.color.senior_color)
    }

    override fun initData(savedInstanceState: Bundle?) {

    }

    override fun addListeners() {
        mBindingView.onClickListener = this

        mBindingView.titleBar.setOnViewLeftClickListener(object : OnViewLeftClickListener {
            override fun leftClick(view: View?) {
                finish()
            }
        })
    }

    override fun onSingleClick(v: View?) {
        when (v?.id) {
            R.id.btn1 -> RouteManager.jump(RoutePath.Senior.URL_TOUCH)
            R.id.btn2 -> RouteManager.jump(RoutePath.Senior.URL_VIEW)
            else -> {
            }
        }
    }
}